src/core/stores/VRange.ts

```diff
@@ -171,17 +258,19 @@ export class VRange {
      * @param position
      * @param reference
      */
-    setStart(reference: VNode, position = RelativePosition.BEFORE): VRange {
-        return VDocument.withRange(() => {
-            reference = reference.firstLeaf();
-            if (position === RelativePosition.AFTER && reference !== this.end) {
-                reference.after(this.start);
-            } else {
-                reference.before(this.start);
-            }
-            this._direction = null; // Invalidate range direction cache.
-            return this;
-        });
+    _setTail(reference: VNode, position = RelativePosition.BEFORE): VRange {
+        reference = reference.firstLeaf();
+        if (!reference.hasChildren() && !reference.properties.atomic) {
```

Comment:
Wait, the concept of atomic did not exist before, is this a hidden fix ?

Reply:
This should be in a commit to set the range in empty non-atomic nodes.